pkg/jx/cmd/step_create_task.go

```diff
@@ -428,9 +437,48 @@ func (o *StepCreateTaskOptions) Run() error {
 		run.Labels = util.MergeMaps(run.Labels, o.labels)
 
 		if o.Verbose {
-			log.Infof("applied tekton CRDs for %s\n", run.Name)
+			log.Infof(" for %s\n", run.Name)
+		}
+	}
+	return nil
+}
+
+// GeneratePipelineActivity generates a initial PipelineActivity CRD so UI/get act can get an earlier notification that the jobs have been scheduled
+func (o *StepCreateTaskOptions) GeneratePipelineActivity(ns string) error {
+
+	build := o.BuildNumber
+	org := o.GitInfo.Organisation
+	repo := o.GitInfo.Name
+	branch := o.Branch
+
+	name := org + "-" + repo + "-" + branch + "-" + build
+	pipeline := org + "/" + repo + "/" + branch
+	log.Infof("PipelineActivity for %s", name)
+	key := &kube.PromoteStepActivityKey{
+		PipelineActivityKey: kube.PipelineActivityKey{
+			Name:     name,
+			Pipeline: pipeline,
+			Build:    build,
+			GitInfo:  o.GitInfo,
+		},
+	}
+
+	if key != nil {
+		jx, _, err := o.JXClientAndDevNamespace()
+		if err != nil {
+			log.Error("error retrieving jxclient")
+			return err
+		}
+		_, _, err = key.GetOrCreate(jx, ns)
```

Comment:
So rather than actually applying the CRD here, we should return it from here, and `GenerateTektonCRDs` should return it with the other CRDs. It should then pass it to `applyPipeline` for the actual applying, same as we do with the other CRDs. That keeps the generation of the CRDs structs separate from creating them in k8s, which makes for cleaner testing, lets the `—no-apply` flag still work, etc.